File: source/Libraries/Thinktecture.AuthorizationServer.Core/Models/Clients.cs
/*
 * Copyright (c) Dominick Baier, Brock Allen.  All rights reserved.
 * see license.txt
 */

using System.Collections.Generic;
using System.Linq;

namespace Thinktecture.AuthorizationServer.Core.Models
{
    public class Clients : List<Client>
    {
        public Client Get(string clientId)
        {
            return (from c in this
                    where c.ClientId.Equals(clientId)
                    select c)
                   .FirstOrDefault();
        }

        public Client ValidateClient(string clientId, string clientSecret)
        {
            // todo: hashing etc

            var client = Get(clientId);
            if (client == null)
            {
                return null;
            }

            if (client.ClientSecret == clientSecret)
            {
                return client;
            }

            return null;
        }
    }
}

File: source/WebHost/Areas/Admin/Scripts/Clients.js
/*
 * Copyright (c) Dominick Baier, Brock Allen.  All rights reserved.
 * see license.txt
 */


$(function () {
    var svc = new as.Service("admin/Clients");

    function Clients(list) {
        this.clients = ko.mapping.fromJS(list);
    }
    Clients.prototype.deleteClient = function (item) {
        var vm = this;
        svc.delete(item.clientId()).then(function () {
            vm.clients.remove(item);
        });
    }

    svc.get().then(function (data) {
        var vm = new Clients(data);
        ko.applyBindings(vm);
    });
});
